test/jdk/java/lang/String/StringRacyConstructor.java

```diff
@@ -0,0 +1,418 @@
+    @MethodSource("strings")
+    public void racyString(String orig) {
+        String racyString = racyStringConstruction(orig);
+        // The contents are indeterminate due to the race
+        assertTrue(validCoder(racyString), orig + " string invalid"
+                + ", racyString: " + inspectString(racyString));
+    }
+
+    @ParameterizedTest
+    @MethodSource("strings")
+    public void racyCodePoint(String orig) {
+        String iffyString = racyStringConstructionCodepoints(orig);
+        // The contents are indeterminate due to the race
+        assertTrue(validCoder(iffyString), "invalid coder in non-deterministic string"
+                + ", orig:" + inspectString(orig)
+                + ", iffyString: " + inspectString(iffyString));
+    }
+
+    @ParameterizedTest
+    @MethodSource("strings")
+    public void racyCodePointSurrogates(String orig) {
+        String iffyString = racyStringConstructionCodepointsSurrogates(orig);
+        // The contents are indeterminate due to the race
+        if (!orig.equals(iffyString))
+            System.err.println("orig: " + orig + ", iffy: " + iffyString + Arrays.toString(iffyString.codePoints().toArray()));
+        assertTrue(validCoder(iffyString), "invalid coder in non-deterministic string"
+                + ", orig:" + inspectString(orig)
+                + ", iffyString: " + inspectString(iffyString));
+    }
+
+    // Test the private methods of StringUTF16 that compress and copy COMPRESSED_STRING
+    // encoded byte arrays.
+    @Test
+    public void verifyUTF16CopyBytes()
+            throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Class<?> stringUTF16 = Class.forName("java.lang.StringUTF16");
+        Method mCompressChars = stringUTF16.getDeclaredMethod("compress",
+                char[].class, int.class, byte[].class, int.class, int.class);
+        mCompressChars.setAccessible(true);
+
+        // First warmup the intrinsic and check 1 case
+        char[] chars = {'a', 'b', 'c', 0xff21, 0xff22, 0xff23};
+        byte[] bytes = new byte[chars.length];
+        int printWarningCount = 0;
+
+        for (int i = 0; i < 1_000_000; i++) {   // repeat to get C2 to kick in
+            // Copy only latin1 chars from UTF-16 converted prefix (3 chars -> 3 bytes)
+            int intResult = (int) mCompressChars.invoke(null, chars, 0, bytes, 0, chars.length);
+            if (intResult == 0) {
+                if (printWarningCount == 0) {
+                    printWarningCount = 1;
+                    System.out.println("StringUTF16.compress returned 0, may not be intrinsic");
+                }
+            } else {
+                assertEquals(3, intResult, "return length not-equal, iteration: " + i);
+            }
+        }
+
+        // Exhaustively check compress returning the correct index of the non-latin1 char.
+        final int SIZE = 48;
+        final byte FILL_BYTE = 0x52;
```

Comment:
```suggestion
        final byte FILL_BYTE = 'R';
```
This makes it more clear that `FILL_BYTE != 'A'` for the logic below.